openstack/compute/v2/extensions/adminactions/requests.go

```diff
@@ -0,0 +1,134 @@
+package adminactions
+
+import "github.com/rackspace/gophercloud"
+
+func actionURL(client *gophercloud.ServiceClient, id string) string {
+	return client.ServiceURL("servers", id, "action")
+}
+
+type CreateBackupOpts struct {
+	// Name: required, name of the backup.
+	Name string
+
+	// BackupType: required, type of the backup, such as "daily".
+	BackupType string
+
+	// Rotation: the number of backups to retain.
+	Rotation int
+}
+
+// ToBackupCreateMap assembles a request body based on the contents of a CreateOpts.
+func (opts CreateBackupOpts) ToCreateBackupMap() (map[string]interface{}, error) {
+	backup := make(map[string]interface{})
+
+	if opts.Name != "" {
+		backup["name"] = opts.Name
+	}
+	if opts.BackupType != "" {
+		backup["backup_type"] = opts.BackupType
+	}
+	backup["rotation"] = opts.Rotation
```

Comment:
Based on the unit tests upstream, it looks like 0 is an acceptable rotation:

https://github.com/openstack/nova/blob/7d4ce9e73f75ac062652a6f500924b1bcec22991/nova/tests/unit/api/openstack/compute/test_create_backup.py#L189-L212

I'll add validation to ensure non-negative integers, though.